docs/design/plugin-registry.md

```diff
@@ -0,0 +1,131 @@
+---
+title: plugin-registry
+authors:
+  - @kaovilai
+reviewers:
+  - @shubham-pampattiwar
+  - @eemcmullan
+  - @savitharaghunathan
+  - @dymurray
+  - @shawn-hurley
+approvers:
+  - @dymurray
+  - @shawn-hurley
+  - @shubham-pampattiwar
+creation-date: 2022-06-14
+last-updated: 2022-06-14
+status: implementable
+---
+
+# Plugin Registry
+
+Plugin registry enhancement replaces current usage of [registry](https://github.com/konveyor/distribution/) [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) insecure [routes](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/networking/configuring-routes) by talking to registry API server inside the velero container itself via openshift-velero-plugin.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+## Open Questions [optional]
+
+Do we want registry logs to be included in velero backup and/or restore logs?
+
+## Summary
+
+removes a listening HTTP server and connect [openshift-velero-plugin](https://github.com/openshift/openshift-velero-plugin/) directly with the [distribution/distribution](https://github.com/distribution/distribution) code to serve requests from [container/image/copy](https://github.com/containers/image/tree/main/copy) method as needed. A "library" to help achieve this is [kaovilai/udistribution](https://github.com/kaovilai/udistribution) which may be moved to konveyor org.
+
+## Motivation
+
+Current image backup workflow depends on registry deployment with exposed insecure routes allowing anyone knowing the URL of the exposed routes to push or pull images to the registry.
+There are some transient errors such as pod health probe timeouts that are resolved by this design.
+This approach will eliminate registry deployment dependency which the operator will no longer have to reconcile.
+
+### Goals
+
+- Exposed routes are eliminated.
+- Remove dependency on registry deployment.
+- Be backward compatible with existing image backup workflow.
+  - images backed up will continue to be in the same directory in object storage as they were prior to this enhancement.
+- Keep deduplication across backups.
+
+### Non-Goals
+
+Creating another custom resource and a controller to replace usage of [distribution/distribution](https://github.com/distribution/distribution).
+
+## Proposal
+
```

Comment:
More from a high level this proposal is to

>Redesign the image backup/restore workflow to instead consume a library directly from within the plugin rather than standing up a transient registry deployment.